### PR TITLE
Fix path issue causing errors instead of exceptions

### PR DESCRIPTION
**Pull request by nmearl**

### CODE REVIEW

File: mosviz/loaders/loader_selection.py [hunk 1]
```diff
@@ -194,12 +194,15 @@ def _validation_checks(self, *args, **kwargs):
             filenames = self.data.get_component(column_name).labels
 
             try:
-                loader(filenames[0])
-            except Exception as e:
+                path = os.sep.join(
+                    self.data._load_log.path.split(os.sep)[:-1])
+                file_path = os.path.join(path, filenames[0])
+                loader(file_path)
```
Comment (astrofrog) [inline mosviz/loaders/loader_selection.py]: I think the ``loader(file_path)`` should maybe be in its own try...except block?

**Comment by nmearl** [comment]
I believe the failure of using relative paths has to do with the fact that on at least one of the new nirspec loaders, I use `Table` to parse the fits file instead of just loading the fits file. In this case, it seems that `Table.read` doesn't like relative paths and was failing.

File: mosviz/loaders/loader_selection.py (hunk 1 above)
Comment (nmearl) [inline mosviz/loaders/loader_selection.py]: Is the latest change alright, or did you have something else in mind.

**Comment by robelgeda** [comment]
Also, some of the images may have a basename of "None" which may fail at this point.

**Comment by nmearl** [comment]
@astrofrog This now includes some fixes to ensure the tests pass. Is this alright with you?

**Pull request merged**